docs/extension-releasing.md

```diff
@@ -1,10 +1,47 @@
 # Extension Releasing
 
-Gemini CLI extensions can be distributed as pre-built binaries through GitHub Releases. This provides a faster and more reliable installation experience for users, as it avoids the need to clone the repository and build the extension from source.
+There are two primary ways of releasing extensions to users:
 
-## Asset naming convention
+- [Git repository](#releasing-through-a-git-repository)
+- [Github Releases](#releasing-through-github-releases)
 
-To ensure Gemini CLI can automatically find the correct release asset for each platform, you should follow this naming convention. The CLI will search for assets in the following order:
+Git repository releases tend to be the simplest and most flexible approach, while GitHub releases can be more efficient on initial install as they are shipped as single archives instead of requiring a git clone which downloads each file individually. Github releases may also contain platform specific archives if you need to ship platform specific binary files.
+
+## Releasing through a git repository
+
+This is the most flexible and simple option. Users can depend on any ref from your git repo, such as a branch or tag, which allows you to manage multiple release channels. You also can point at any git repository, so your extension does not have to be hosted on github.
+
+For instance, you can maintain a `stable` branch, which users can install this way `gemini extensions install <your-repo-uri> --ref=stable`. Or, you could make this the default by treating your default branch as your stable release, and doing development in a different branch (for instance called `dev`). You can maintain as many branches or tags as you like, providing maximum flexibility for you and your users.
```

Comment:
@chrstnb should we consider using the `stable` ref by default if it exists? This would allow people to use their main branch for development while not risking  users accidentally depending on it (otherwise they have to do `--ref=stable` to be on the stable release channel).

In particular this can avoid issues around having to force push to your default branch if your history starts to diverge from your development branch due to cherry picks etc.

Reply:
I think the idea is that the github repo based approach is super simple and accessible. I'm wary of introducing a specific way of doing things yet.

Reply:
I am fine with that, I do worry this would be hard to change in the future though.